discovery/gossiper_test.go

```diff
@@ -734,9 +734,11 @@ func createTestCtx(startHeight uint32) (*testCtx, func(), error) {
 
 			return nil
 		},
-		NotifyWhenOnline: func(target *btcec.PublicKey,
+		NotifyWhenOnline: func(target [33]byte,
 			peerChan chan<- lnpeer.Peer) {
-			peerChan <- &mockPeer{target, nil, nil}
+
+			pk, _ := btcec.ParsePubKey(target[:], btcec.S256())
+			peerChan <- &mockPeer{pk, nil, nil}
```

Comment:
Have you looked into also making the `mockPeer` use a `[33]byte` as well?

Reply:
Yes, it was a little bit more involved and I didn't want to broaden the scope of this PR. I can make an attempt in another PR.